syncer/peers/nofork/service.go

```diff
@@ -0,0 +1,131 @@
+package nofork
+
+import (
+	"context"
+
+	"github.com/0xPolygon/polygon-edge/network/grpc"
+	"github.com/0xPolygon/polygon-edge/syncer/peers"
+	"github.com/0xPolygon/polygon-edge/syncer/peers/nofork/proto"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+const (
+	SyncPeerServiceLoggerName = "sync-peer-service"
+)
+
+type SyncPeerService struct {
+	proto.UnimplementedSyncPeerServer
+
+	logger     hclog.Logger
+	blockchain Blockchain
+	network    Network
+
+	id           string // node ID
+	peerStatusCh chan *NoForkPeer
+}
```

Comment:
What is the point of redefining this here in a separate package?

You need to drag around references to the blockchain and network modules, and you fall into a situation where your parent class has a reference to you, but both of you have these interface references as well 🤷‍♂️

Reply:
I don't see what you're trying to say. The struct will have the references it needs